### PR TITLE
Only depend on sympy 1.12 for conda (no 3.13 there anyways)

### PR DESCRIPTION
**Pull request by albanD**
Fixing nightly after https://github.com/pytorch/pytorch/pull/130895

### CODE REVIEW

File: .circleci/scripts/binary_linux_test.sh [hunk 1]
```diff
@@ -52,13 +52,6 @@ if [[ "\$python_nodot" = *39* ]]; then
   NUMPY_PIN=">=1.20"
 fi
 
-if [[ "\$python_nodot" = *38* ]]; then
-  # sympy 1.12.1 is the last version that supports Python 3.8
-  SYMPY_PIN="==1.12.1"
```
Comment (XuehaiPan) [inline .circleci/scripts/binary_linux_test.sh:57]: You could search and replace the occurance of `1.2.1` and `1.3.0`.

Comment (albanD) [reply .circleci/scripts/binary_linux_test.sh:57]: I only want to change the conda ones though, I do want pypi to be on these newer versions

**Comment by albanD** [comment]
Also @XuehaiPan the optree failures are also related to the upgrade you did. Is there no binary release of optree for 3.13 atm?

**Comment by XuehaiPan** [comment]
> Also @XuehaiPan the optree failures are also related to the upgrade you did. Is there no binary release of optree for 3.13 atm?

FYI, there is no conda package for optree on anaconda main channel. The latest optree for Python 3.13 support is installable from sdist. The binary wheel of optree for Python 3.13 is not available yet because Python 3.13 is currently not GA.

File: requirements.txt [hunk 2]
```diff
@@ -20,4 +20,4 @@ ninja
 # setuptools was removed from default python install
 setuptools ; python_version >= "3.12"
 packaging
-optree>=0.12.0
+optree>=0.12.0 ; python_version <= "3.12"
```
Comment (albanD) [inline requirements.txt:23]: FYI @XuehaiPan reverting this as well then as we cannot install it for 3.13 binaries testing as it is only available as an sdist

**Review by atalman — APPROVED** [review]
lgtm

**Pull request closed without merging**